### PR TITLE
Hide `loader.entrypoint` manifest option

### PR DESCRIPTION
**Pull request by dimakuv**
For end users, the `loader.entrypoint` manifest option is always the LibOS binary. Similarly, `sgx.trusted_files` must always contain the LibOS binary.

This commit is a counterpart to the [commit "[Python,CI-Examples] Hide loader.entrypoint manifest option"](https://github.com/gramineproject/gramine/pull/1716) in the core Gramine repo.

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)

### CODE REVIEW

**Review by mkow — APPROVED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @dimakuv)

___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/gramineproject/examples/99#-Nwy7b6oCLbFUkBS72eJ):*




_Suggestion:_
```
[python,CI-Examples] Hide
```

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow)

___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/gramineproject/examples/99#-Nwy7b6oCLbFUkBS72eJ):*
Done.

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NwyTdWq-IjR8iNQYyCH:bx8w2vb) (no related file):*
But it is clearly mentioned in our README -- "These examples are tested only against the most recent Gramine release (i.e., these examples are not guaranteed to work with older releases and unreleased versions of Gramine, including the latest master branch commits of Gramine).", does this imply that we need to hold this PR?

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NwyUg3PCd6tCpRHp_0K:b-8b1e79) (no related file):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

But it is clearly mentioned in our README -- "These examples are tested only against the most recent Gramine release (i.e., these examples are not guaranteed to work with older releases and unreleased versions of Gramine, including the latest master branch commits of Gramine).", does this imply that we need to hold this PR?
</blockquote></details>

Hm, good point. I'm not sure what our policy should be. Technically, @kailun-qin is right, and as is currently written in the README, we should postpone this PR until right-before the next release.

Alternatively, we can change our README policy to be simpler: `master` Examples are guaranteed to work with `master` Gramine. If one wants to use a release of Gramine, they should use the same release in Examples. This is similar to how we tie GSC to Gramine. @woju @mkow @kailun-qin WDYT?

___

**Review by kailun-qin — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NwyXMQH4RDJJOsxzBj9:b-l1d8j1) (no related file):*
> Alternatively, we can change our README policy to be simpler: `master` Examples are guaranteed to work with `master` Gramine. If one wants to use a release of Gramine, they should use the same release in Examples. This is similar to how we tie GSC to Gramine.

+1. I think we validate Examples on the corresponding Gramine before each release, so it should be fine that we state this way.

___

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-Nx-V3g1BwHP2zNLMECg:b32v57) (no related file):*
<details><summary><i>Previously, kailun-qin (Kailun Qin) wrote…</i></summary><blockquote>

> Alternatively, we can change our README policy to be simpler: `master` Examples are guaranteed to work with `master` Gramine. If one wants to use a release of Gramine, they should use the same release in Examples. This is similar to how we tie GSC to Gramine.

+1. I think we validate Examples on the corresponding Gramine before each release, so it should be fine that we state this way.
</blockquote></details>

+1

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NxBqdhk5pbLelyHZKb4:be5j0ud) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

+1
</blockquote></details>

Created https://github.com/gramineproject/examples/pull/101, please check it.

___

**Review by woju — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @dimakuv and @kailun-qin)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NxHYlFADyTkxlC6GaNX:bnrxknh) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Created https://github.com/gramineproject/examples/pull/101, please check it.
</blockquote></details>

This would be bad, because newbie users (those, who need examples in the first place) will install from packages or use docker image, and then will checkout this repo. So the repo should stay as is I think.

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @kailun-qin and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NxI2nn42bW-VqJuzFTB:bkfei5u) (no related file):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

This would be bad, because newbie users (those, who need examples in the first place) will install from packages or use docker image, and then will checkout this repo. So the repo should stay as is I think.
</blockquote></details>

@woju What do you mean exactly? That we don't use this PR and thus don't remove `loader.entrypoint` from these examples?

At which point do you propose to resurrect this PR? After the next version of Gramine is released?

___

**Review by woju — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @dimakuv and @kailun-qin)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NxI_umu1R0Wok0nj9M9:bwqe1eb) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@woju What do you mean exactly? That we don't use this PR and thus don't remove `loader.entrypoint` from these examples?

At which point do you propose to resurrect this PR? After the next version of Gramine is released?
</blockquote></details>

Yes, after we release (package) something that allows to remove `loader.entrypoint`. Rationale is that novice users will install from packages, and then checkout the default branch (`HEAD`) of this repo.

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion (waiting on @kailun-qin and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-NxLr1_r3vHD2D93U55F:bwqtvc0) (no related file):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Yes, after we release (package) something that allows to remove `loader.entrypoint`. Rationale is that novice users will install from packages, and then checkout the default branch (`HEAD`) of this repo.
</blockquote></details>

During the [Gramine Tuesday meeting](https://github.com/gramineproject/gramine/discussions/1871), it was decided to keep the current Examples policy as-is, and postpone merging this breaking change until after the next Gramine release is done.

In other words, we will follow @woju's advice.

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion (waiting on @kailun-qin and @woju)

*[a discussion](https://reviewable.io/reviews/gramineproject/examples/99#-NwyTdWp27gz3QgF5NFZ:-O9s4aSU8v0rhOOSvrIR:b-pgoye5) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

During the [Gramine Tuesday meeting](https://github.com/gramineproject/gramine/discussions/1871), it was decided to keep the current Examples policy as-is, and postpone merging this breaking change until after the next Gramine release is done.

In other words, we will follow @woju's advice.
</blockquote></details>

New release of Gramine (v1.8) was done: https://github.com/gramineproject/gramine/releases/tag/v1.8

If I understand our policy correctly, now we can merge this PR, and afterwards I'll tag the Examples repo with `v1.8` tag.

___

**Review by kailun-qin — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: all files reviewed, 1 unresolved discussion (waiting on @woju)

**Review by woju — APPROVED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by dimakuv — COMMENTED** [review]
> Reviewed 5 of 11 files at r1, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/99)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved
